OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,475 @@
+                else
+                {
+                    return KeyCode.None;
+                }
+            }
+            else
+            {
+                string changedKey = key;
+                if (key == "control" || key == "ctrl")
+                {
+                    return KeyCode.LeftControl;
+                }
+                else if (key == "shift")
+                {
+                    return KeyCode.LeftShift;
+                }
+                else if (key == "alt")
+                {
+                    return KeyCode.LeftAlt;
+                }
+                KeyCode code = (KeyCode)Enum.Parse(typeof(KeyCode), changedKey, true);
+                if (Enum.IsDefined(typeof(KeyCode), code))
+                {
+                    return code;
+                }
+                else
+                {
+                    return KeyCode.None;
+                }
+            }
+        }
+
+        private Int64 ParseCombination(string combo, bool forRemoval = false)
+        {
+            combo = combo.Trim();
+            int[] thisCombination = new int[_maxComboLength];
+            int i = 0;
+            foreach (string key in combo.Split('+'))
+            {
+                if (i >= _maxComboLength)
+                    return (int)RegistrationCode.CombinationTooLong;
+                KeyCode code = StringToKeyCode(key);
+                if (code != KeyCode.None)
+                {
+                    thisCombination[i] = (int)code;
+                }
+                else
+                {
+                    return (int)RegistrationCode.InvalidCombination;
+                }
+                i++;
+            }
+            Array.Sort(thisCombination);
+            Int64 hash = 0;
+            for (i = 0; i < _maxComboLength; i++)
+            {
+                hash = hash * _maxUsefulKey + thisCombination[i];
+            }
+            if (hash < _maxUsefulKey && _gameBindingCounter[hash] > 0)
+            {
+                return (int)RegistrationCode.CombinationTaken;
+            }
+            return (_comboRegistry.ContainsKey(hash) && !forRemoval ? (int)RegistrationCode.CombinationTaken : hash);
+        }
+
+        public RegistrationCode RegisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            castCombo.ClearSingles();
+            if (castCombo == null || castCombo.Combo == null)
+            {
+                _console.WriteLine("combination is null");
+                return RegistrationCode.InvalidCombination;
+            }
+            string[] combs = castCombo.Combo.ToLower().Split('/');
+            List<Int64> combos = new List<Int64>();
+            foreach (string comstr in combs)
+            {
+                Int64 hash = ParseCombination(comstr);
+                if (hash <= 0)
+                {
+                    return (RegistrationCode)hash;
+                }
+                else
```

Comment:
This else can be removed, since you return in the if

Reply:
done